cmd/thanos/receive.go

```diff
@@ -303,6 +303,11 @@ func runReceive(
 			Help: "Number of Multi DB completed reloads with flush and potential upload due to hashring changes",
 		})
 
+		level.Debug(logger).Log("msg", "cleaning storage lock files")
+		if err := dbs.Clean(); err != nil {
```

Comment:
Looks scary! ;p 

Also hm....... I think this needs explanation in the flag, that lock file is removed on the start always.

NOTE: Maybe worth to add log msg if we have lock file still? (indicating hard shutdown)

Reply:
Also we need the same on ruler for now

Reply:
```suggestion
		if err := dbs.RemoveLockFilesIfAny(); err != nil {
```